src/infrastructure/storage/gcs-sync.ts

```
@@ -0,0 +1,176 @@
+    } catch (error) {
+      console.error("❌ Error downloading store:", error);
+      throw error;
+    }
+  }
+
+  /**
+   * Upload the entire store directory from local filesystem to GCS
+   * Overwrites existing files in GCS
+   */
+  async uploadStore(localStorePath: string): Promise<void> {
+    console.log(
+      `📤 Uploading store from ${localStorePath}/ to gs://${this.bucketName}/${this.remotePath}/`,
+    );
+
+    if (!fs.existsSync(localStorePath)) {
+      console.log("No local store directory found, skipping upload");
+      return;
+    }
+
+    const bucket = this.storage.bucket(this.bucketName);
+    let uploadCount = 0;
+
+    try {
+      // Recursively upload all files
+      uploadCount = await this.uploadDirectory(
+        localStorePath,
+        localStorePath,
+        bucket,
+      );
+      console.log(`📤 Store upload complete: ${uploadCount} files uploaded`);
+    } catch (error) {
+      console.error("❌ Error uploading store:", error);
+      throw error;
+    }
+  }
+
+  private async uploadDirectory(
+    currentPath: string,
+    basePath: string,
+    bucket: Bucket,
+  ): Promise<number> {
+    const items = fs.readdirSync(currentPath);
+    let uploadCount = 0;
+
+    for (const item of items) {
+      const fullPath = path.join(currentPath, item);
+      const stat = fs.statSync(fullPath);
+
+      if (stat.isDirectory()) {
+        uploadCount += await this.uploadDirectory(fullPath, basePath, bucket);
+      } else {
+        // Calculate relative path for GCS
+        const relativePath = path.relative(basePath, fullPath);
+        const gcsPath = `${this.remotePath}/${relativePath.replace(/\\/g, "/")}`;
+
+        // Upload file
+        await bucket.upload(fullPath, {
+          destination: gcsPath,
+          metadata: {
+            contentType: fullPath.endsWith(".json")
+              ? "application/json"
+              : "application/octet-stream",
```

Interesting, do we expect non json here? otherwise I'd be tempted to error out if we the file doesnt end with .json, rather than silently continuing

---

agree, we should only expect json here.